pulp_ansible/app/viewsets.py

```diff
@@ -98,6 +99,28 @@ class RoleViewSet(ContentViewSet):
     serializer_class = RoleSerializer
     filterset_class = RoleFilter
 
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_repository_model_or_obj_perms:ansible.view_role",
```

Comment:
You don't need a condition here, Roles is a content type and doesn't have permissions directly assigned to them.